worker/gouging.go

```diff
@@ -16,62 +17,89 @@ const keyGougingChecker contextKey = "GougingChecker"
 
 type (
 	GougingChecker interface {
-		Check(rhpv2.HostSettings) GougingResults
+		CheckHS(*rhpv2.HostSettings) GougingResults
+		CheckPT(*rhpv3.HostPriceTable, types.Currency, uint64, uint64) GougingResults
 	}
 
 	GougingResults struct {
-		downloadErr     error
-		formContractErr error
-		uploadErr       error
+		downloadErr error
+		gougingErr  error
+		uploadErr   error
 	}
 
 	gougingChecker struct {
-		settings    api.GougingSettings
-		minShards   int
-		totalShards int
+		settings   api.GougingSettings
+		redundancy api.RedundancySettings
 	}
 
 	contextKey string
 )
 
-func PerformGougingChecks(ctx context.Context, hs rhpv2.HostSettings) GougingResults {
-	if gc, ok := ctx.Value(keyGougingChecker).(GougingChecker); ok {
-		return gc.Check(hs)
+var _ GougingChecker = gougingChecker{}
+
+func PerformGougingChecks(ctx context.Context, hs *rhpv2.HostSettings, pt *rhpv3.HostPriceTable) (results GougingResults) {
+	gc, ok := ctx.Value(keyGougingChecker).(GougingChecker)
+	if !ok {
+		panic("no gouging checker attached to the context") // developer error
 	}
-	panic("no gouging checker attached to the context") // developer error
+
+	results.merge(gc.CheckHS(hs))
+	results.merge(gc.CheckPT(pt, types.ZeroCurrency, 0, 0))
```

Comment:
This is a bit sad perhaps, I think maybe a better solution is to extract those checks (txn fee, period and window size) and not consider them gouging checks maybe? But just have them be some extra checks we perform in the autopilot.

`PerformGougingChecks` is called from the `worker` so either we have to ask those values from the autopilot or not perform those checks in the worker on-the-fly

Reply:
Couple of options actually:
- we can check these settings in the autopilot, not re-check in the worker
- we can fetch these values on the fly from the autopilot
- we can store these values on the bus
- we can send this information along with form/renew requests
- we can check on `period+renewwindow` (since that info is in the contract's new end height, but if it fires we won't know whether the period or renew window setting did the trick, although we can probably deduce what the issue was if we both log them)

I'm personally in favour of checking everywhere:
- txn fee
- block height 

only checking these values in the autopilot:
- max period
- renew windowd

just to keep things simple and to prevent having to pass along this information.